### DISCUSSION TITLE
How do I host this without a server?

### DISCUSSION THREAD

**Discussion by FreezingChima** [searxng/searxng#1759] [Q&A] [answered]
I would like to not have to use docker to host this, or use a server. I also don't understand how to use the source code to customize the pages when there isn't an html file for me to edit... :/

I just wanted to host this on my github site or something.

**Comment by tiekoetter** [comment, marked as answer]
GitHub Pages doesn't allow this.

You could try this: https://github.com/searxng/searxng/discussions/976

If you want to edit the HTML: https://github.com/searxng/searxng/tree/master/searx/templates/simple

**Reply by FreezingChima** [reply]
Hold on... the html page files when opened in a web browser are just lines of code...
I just want the full program with complete pages. They literally have no CSS.

This is results.html:
![image](https://user-images.githubusercontent.com/112496936/189486626-d906ee98-9ee0-4ae9-845c-cd496723e937.png)
???

**Reply by return42** [reply]
You asked two questions, and @tiekoetter answered both

First question:

>>  I would like to not have to use docker to host this, or use a server. 

> GitHub Pages doesn't allow this.
> You could try this: https://github.com/searxng/searxng/discussions/976

Second question:

>>> I also don't understand how to use the source code to customize the pages

>> If you want to edit the HTML: https://github.com/searxng/searxng/tree/master/searx/templates/simple

> the html page files when opened in a web browser are just lines of code...

These are the HTML templates from where the "complete pages" are rendered.  Without deeper developer knowledge you cant customize the HTML output of SearXNG.

To answer both questions in one: SearXNG is a server software, not a static HTML page.

**Reply by FreezingChima** [reply]
Are there any good alternatives to SearX and SearXNG that are just html applications?

**Reply by tiekoetter** [reply]
@FreezingChima I think you have a misunderstanding on what the difference between a static and dynamic web application is.

To answer your question there is no static alternative to SearX/NG.

You can always use one of the public instances at https://searx.space
